subprojects/client-protocol/src/main/java/org/gradle/profiler/client/protocol/agent/AgentClientContainer.java

```diff
@@ -0,0 +1,23 @@
+package org.gradle.profiler.client.protocol.agent;
+
+import org.gradle.profiler.client.protocol.Client;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+public enum AgentClientContainer {
+    INSTANCE;
+
+    private final AtomicReference<Client> client = new AtomicReference<>();
```

Comment:
I don't think an `AtomicReference` is the right thing here. Why not use a `volatile Client client` field?

Reply:
Hmm, I think there is no big difference, except that AtomicReference might take more memory. I set it to volatile though.

Reply:
In your usage it doesn't make a difference, right. When looking at this I was wondering why we did the extra locking when updating the atomic reference, and then I thought it would be better to use `volatile` to remove the confusion.